src/extensions/idle.rs

```diff
@@ -26,10 +35,50 @@ use std::time::Duration;
 #[derive(Debug)]
 pub struct Handle<'a, T: Read + Write + 'a> {
     session: &'a mut Session<T>,
+    unsolicited_responses_tx: mpsc::Sender<UnsolicitedResponse>,
     keepalive: Duration,
     done: bool,
 }
 
+
+/// 'IdleIterator' allows a client to iterate over unsolicited responses during an IDLE operation.
+///
+/// As long as a [`IdleIterator`] is active, the mailbox cannot be otherwise accessed.
+pub struct IdleIterator<'a, T: Read + Write + 'a> {
+    handle: Handle<'a, T>,
+    buffer: Vec<u8>,
+    keepalive: bool,
+}
+
+impl<'a, T: Read + Write + 'a> IdleIterator<'a, T> {
+    fn new(handle: Handle<'a, T>, keepalive: bool) -> IdleIterator<'a, T> {
+        IdleIterator { handle, buffer: Vec::new(), keepalive }
+    }
+}
+
+/// 'TimeoutIdleIterator' allows a client to iterate over unsolicited responses during an IDLE operation.
+///
+/// As long as a [`TimeoutIdleIterator`] is active, the mailbox cannot be otherwise accessed.
+pub struct TimeoutIdleIterator<'a, T: SetReadTimeout + Read + Write + 'a> {
```

Comment:
Why do we need a separate type for this?

Reply:
Because the if a timeout is used, dropping the Iterator resets the timeout and this can't be done in IdleIterator if T isn't SetReadTimeout. So that requires different Drop implementations depending if T's traits. Or maybe it could be done in IdleIterator by registering a callback, but I'm not sure it's cleaner.

Reply:
Ah, yes, that makes sense... Fair enough!